hummingbot/connector/gateway/clob_spot/data_sources/kujira/kujira_api_data_source.py

```diff
@@ -0,0 +1,992 @@
+from hummingbot.core.utils.async_utils import safe_ensure_future, safe_gather
+
+from ..gateway_clob_api_data_source_base import GatewayCLOBAPIDataSourceBase
+from .kujira_constants import (
+    CONNECTOR,
+    DELAY_BETWEEN_RETRIES,
+    KUJIRA_NATIVE_TOKEN,
+    MARKETS_UPDATE_INTERVAL,
+    NUMBER_OF_RETRIES,
+    TIMEOUT,
+    UPDATE_ORDER_STATUS_INTERVAL,
+)
+from .kujira_helpers import (
+    AsyncLock,
+    automatic_retry_with_timeout,
+    convert_market_name_to_hb_trading_pair,
+    generate_hash,
+)
+from .kujira_types import OrderStatus as KujiraOrderStatus
+
+
+class KujiraAPIDataSource(GatewayCLOBAPIDataSourceBase):
+
+    def __init__(
+        self,
+        trading_pairs: List[str],
+        connector_spec: Dict[str, Any],
+        client_config_map: ClientConfigAdapter,
+    ):
+        super().__init__(
+            trading_pairs=trading_pairs,
+            connector_spec=connector_spec,
+            client_config_map=client_config_map
+        )
+
+        self._chain = connector_spec["chain"]
+        self._network = connector_spec["network"]
+        self._connector = CONNECTOR
+        self._owner_address = connector_spec["wallet_address"]
+        self._payer_address = self._owner_address
+
+        self._trading_pair = None
+        if self._trading_pairs:
+            self._trading_pair = self._trading_pairs[0]
+
+        self._markets = None
+        self._market = None
+
+        self._user_balances = None
+
+        self._tasks = DotMap({
+            "update_order_status_loop": None
+        }, _dynamic=False)
+
+        self._locks = DotMap({
+            "place_order": AsyncLock(),
+            "place_orders": AsyncLock(),
+            "cancel_order": AsyncLock(),
+            "cancel_orders": AsyncLock(),
+            "settle_market_funds": AsyncLock(),
+            "settle_markets_funds": AsyncLock(),
+            "settle_all_markets_funds": AsyncLock(),
+            "all_active_orders": AsyncLock(),
+        }, _dynamic=False)
+
+        self._gateway = GatewayHttpClient.get_instance(self._client_config)
+
+        self._all_active_orders = None
+
+        self._snapshots_min_update_interval = 30
+        self._snapshots_max_update_interval = 60
+        self.cancel_all_orders_timeout = TIMEOUT
+
+    @property
+    def connector_name(self) -> str:
+        return CONNECTOR
+
+    @property
+    def real_time_balance_update(self) -> bool:
+        return False
+
+    @property
+    def events_are_streamed(self) -> bool:
+        return False
+
+    @staticmethod
+    def supported_stream_events() -> List[Enum]:
+        return [
+            MarketEvent.TradeUpdate,
+            MarketEvent.OrderUpdate,
+            MarketEvent.OrderFilled,
+            AccountEvent.BalanceEvent,
+            OrderBookDataSourceEvent.TRADE_EVENT,
+            OrderBookDataSourceEvent.DIFF_EVENT,
+            OrderBookDataSourceEvent.SNAPSHOT_EVENT,
+        ]
+
+    def get_supported_order_types(self) -> List[OrderType]:
```

Comment:
why we only support limit if in the constants we have Market too? will be good to have the two of them

Reply:
I've added the support to market orders here as well now, and updated our enums to remove non-used types.
But we usually use only limit orders.